src/main/java/com/thebois/models/world/World.java

```diff
@@ -179,6 +163,50 @@ private void postObstacleEvent(final int posX, final int posY) {
         Pawntastic.BUS.post(obstacleEvent);
     }
 
+    @Override
+    public Optional<IStructure> findNearestStructure(final Position position) {
+        final int searchRow = Math.round(position.getPosY());
+        final int searchCol = Math.round(position.getPosX());
+
+        final int maxSearchRadius = this.worldSize / 2;
+
+        final Collection<IStructure> foundStructures = MatrixUtils.matrixSpiralSearch(
+            this.structureMatrix,
+            searchRow,
+            searchCol,
+            maxSearchRadius,
+            1);
+
+        if (foundStructures.iterator().hasNext()) {
+            return Optional.of(foundStructures.iterator().next());
+        }
+        else {
+            return Optional.empty();
+        }
+    }
+
+    @Override
+    public Optional<IStructure> findNearestIncompleteStructure(final Position position) {
+        final int searchRow = Math.round(position.getPosY());
+        final int searchCol = Math.round(position.getPosX());
+
+        final int maxSearchRadius = this.worldSize;
+
+        final Collection<IStructure> foundStructures =
+            MatrixUtils.matrixSpiralSearch(this.structureMatrix,
+                                           searchRow,
+                                           searchCol,
+                                           maxSearchRadius,
+                                           10);
+
+        for (final IStructure structure : foundStructures) {
+            if (!structure.isCompleted()) {
+                return Optional.of(structure);
+            }
+        }
+        return Optional.empty();
```

Comment:
These to methods are almost identical, only difference are some values, make one method maybe?